dotnet/src/SemanticKernel.Abstractions/AI/FunctionChoiceBehaviors/FunctionChoiceBehavior.cs

```diff
@@ -0,0 +1,187 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.SemanticKernel;
+
+/// <summary>
+/// Represents the base class for different function choice behaviors.
+/// </summary>
+[JsonPolymorphic(TypeDiscriminatorPropertyName = "type")]
+[JsonDerivedType(typeof(AutoFunctionChoiceBehavior), typeDiscriminator: "auto")]
+[JsonDerivedType(typeof(RequiredFunctionChoiceBehavior), typeDiscriminator: "required")]
+[JsonDerivedType(typeof(NoneFunctionChoiceBehavior), typeDiscriminator: "none")]
+public abstract class FunctionChoiceBehavior
```

Comment:
Can we evaluate how this adapts to requirements for assistant-agent _tool-choice_?

https://platform.openai.com/docs/api-reference/runs/createRun#runs-createrun-tool_choice

I wonder if there's a way to not bind this base abstraction so tightly to chat-completion only?  Maybe there is an even more generic base abstraction?  The current form might not adapt so well for assistants.

Reply:
Why would we not / are we not using the same content types for agents?